py4cast/datasets/base.py

```diff
@@ -542,6 +612,207 @@ def get_param_list(
+
+                loutputs.append(tmp_state)
+                tmp_state = NamedTensor(
+                    tensor=tensor[: self.settings.num_input_steps],
+                    **deepcopy(state_kwargs),
+                )
+                linputs.append(tmp_state)
+
+        lforcings = generate_forcings(
+            date=self.timestamps.datetime,
+            output_terms=self.output_timestamps.terms,
+            grid=self.grid,
+        )
+
+        for forcing in lforcings:
+            forcing.unsqueeze_and_expand_from_(linputs[0])
+
+        return Item(
+            inputs=NamedTensor.concat(linputs),
+            outputs=NamedTensor.concat(loutputs),
+            forcing=NamedTensor.concat(lforcings),
+        )
+
+    def plot(self, item: Item, step: int, save_path: Path = None) -> None:
+        # Retrieve the named tensor
+        ntensor = item.inputs if step <= 0 else item.outputs
+
+        # Retrieve the timestep data index
+        if step <= 0:  # input step
+            index_tensor = step + self.settings.num_input_steps - 1
+        else:  # output step
+            index_tensor = step - 1
+
+        # Sort parameters by level, to plot each level on one line
+        levels = sorted(list(set([p.level for p in self.params])))
+        dict_params = {level: [] for level in levels}
+        for param in self.params:
+            if param.parameter_short_name in ntensor.feature_names:
+                dict_params[param.level].append(param)
+
+        # Groups levels 0m, 2m and 10m on one "surf" level
+        dict_params["surf"] = []
+        for lvl in [0, 2, 10]:
+            if lvl in levels:
+                dict_params["surf"] += dict_params.pop(lvl)
+
+        # Plot settings
+        kwargs = {"projection": self.grid.projection}
+        nrows = len(dict_params.keys())
+        ncols = max([len(param_list) for param_list in dict_params.values()])
+        fig, axs = plt.subplots(nrows, ncols, figsize=(20, 15), subplot_kw=kwargs)
+
+        for i, level in enumerate(dict_params.keys()):
+            for j, param in enumerate(dict_params[level]):
+                pname = param.parameter_short_name
+                tensor = ntensor[pname][index_tensor, :, :, 0]
+                arr = tensor.numpy()[::-1]  # invert latitude
+                vmin, vmax = self.stats[pname]["min"], self.stats[pname]["max"]
+                img = axs[i, j].imshow(
+                    arr, vmin=vmin, vmax=vmax, extent=self.grid.grid_limits
+                )
+                axs[i, j].set_title(pname)
+                axs[i, j].coastlines(resolution="50m")
+                cbar = fig.colorbar(img, ax=axs[i, j], fraction=0.04, pad=0.04)
+                cbar.set_label(param.unit)
+
+        plt.suptitle(
+            f"Run: {self.timestamps.datetime} - Valid time: {self.timestamps.validity_times[step]}"
+        )
+        plt.tight_layout()
+
+        if save_path is not None:
+            plt.savefig(save_path)
+            plt.close()
```

Comment:
Maybe add a comment to explain why me might do this (gif building below) otherwise we wonder "why do you plot since you don't save and don't show ?"